python/src/scipp/plotting/resampling_model.py

```diff
@@ -236,18 +231,29 @@ def _resample(self, array):
         # We could bin with all edges and then use `bins.sum()` but especially
         # for inputs with many bins handling the final edges using `histogram`
         # is faster with the current implementation of `sc.bin`.
-        edges = self.edges[-1]
-        dim = edges.dims[-1]
+        # To avoid excessive memory use by `bin` we have to ensure that the
+        # dimension handled by `histogram` is not a dimension with many bins.
+        # We therefore select the smallest dimension of the input array for
+        # handling with `histogram`.
+        sizes = {}
+        for edges in self.edges:
+            dim = edges.dims[-1]
+            sizes[dim] = array.sizes[dim]
+        dim = min(sizes, key=sizes.get)
         # `bin` applies masks, but later we add rebinned masks. This would be
         # inconsistent with how dense data is handled, where data is preserved
         # even if masked, but masks grow. Therefore, we remove masks here. They
         # get handled in _call_resample.
         array = self._strip_masks(array)
         if dim in array.bins.coords:
```

Comment:
The `dim` defined on line 242 is being overwritten here. I think this is not good?

Reply:
Where does it get overwritten?

Reply:
Sorry, forget what i wrote, i read `for dim in array.bins.coords:` instead of `if dim in array.bins.coords:`